equinix/resource_fabric_service_profile.go

```diff
@@ -936,22 +929,20 @@ func accessPointColoFabricSpToTerra(accessPointColol []v4.ServiceProfileAccessPo
 	return mappedAccessPointColol
 }
 
-func serviceMetroFabricSpToTerra(serviceMetrol []v4.ServiceMetro) []map[string]interface{} {
+func serviceMetroFabricSpToTerra(serviceMetrol []v4.ServiceMetro) []interface{} {
```

Comment:
If the method input is a slice, then we need to keep the for loop. We will add map[string]interface{} elements to a []interface{} made with the right length by index. Then return the []interface{} directly. We won't need to use append().